spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -383,14 +390,52 @@ public boolean dropNamespace(String[] namespace) throws NoSuchNamespaceException
 
   @Override
   public final void initialize(String name, CaseInsensitiveStringMap options) {
-    this.cacheEnabled = Boolean.parseBoolean(options.getOrDefault("cache-enabled", "true"));
-    Catalog catalog = buildIcebergCatalog(name, options);
+    this.cacheEnabled = PropertyUtil.propertyAsBoolean(options,
+        CatalogProperties.TABLE_CACHE_ENABLED, CatalogProperties.TABLE_CACHE_ENABLED_DEFAULT);
+
+    // If the user disabled caching and did not set the cache.expiration-interval-ms, we'll set it to zero for
+    // them on their behalf. If they disabled caching but explicitly set a non-zero cache expiration
+    // interval, we will fail initialization as that's an invalid configuration.
+    long defaultCacheExpirationInterval =
+        cacheEnabled ? CatalogProperties.TABLE_CACHE_EXPIRATION_INTERVAL_MS_DEFAULT : 0L;
+
+    this.cacheExpirationIntervalMs = PropertyUtil.propertyAsLong(options,
+        CatalogProperties.TABLE_CACHE_EXPIRATION_INTERVAL_MS,
+        defaultCacheExpirationInterval);
+
+    // Normalize usage of -1 to 0 as we'll call Duration.ofMillis on this value.
+    if (cacheExpirationIntervalMs < 0) {
```

Comment:
I could see this being an error case. Why would you sent the interval to a negative value?

Reply:
To allow for `-1` to mean off. Symmetry with Spark.

We don't need to keep that though.

Reply:
I'm fine with that. Negative values turn off caching, just like 0 does then?

Reply:
No negative values would leave caching on, just disable cache expiration.